dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs

```diff
@@ -140,15 +140,159 @@ private protected ClientBase(ILoggerFactory? loggerFactory = null)
 
         ValidateMaxTokens(textRequestSettings.MaxTokens);
 
-        var options = CreateCompletionsOptions(text, textRequestSettings);
+        var options = this.CreateCompletionsOptions(text, textRequestSettings);
 
-        Response<StreamingCompletions>? response = await RunRequestAsync<Response<StreamingCompletions>>(
-            () => this.Client.GetCompletionsStreamingAsync(this.DeploymentOrModelName, options, cancellationToken)).ConfigureAwait(false);
+        StreamingResponse<Completions>? response = await RunRequestAsync<StreamingResponse<Completions>>(
+            () => this.Client.GetCompletionsStreamingAsync(options, cancellationToken)).ConfigureAwait(false);
 
-        using StreamingCompletions streamingChatCompletions = response.Value;
-        await foreach (StreamingChoice choice in streamingChatCompletions.GetChoicesStreaming(cancellationToken))
+        // Keep running the request stream in the background populating the choices cache
+        // This will keep the behavior similar to previous versions of the Azure SDK allowing return of Choices to be iterated over
+        var cachedChoices = new Dictionary<int, List<Choice>>();
+        var results = new List<TextStreamingResult>();
+        bool streamingStarted = false;
+        bool streamingEnded = false;
+
+        _ = Task.Run(async () =>
+        {
+            await foreach (Completions completions in response)
+            {
+                // first yield;
+                foreach (var choice in completions.Choices)
+                {
+                    if (!cachedChoices.ContainsKey(choice.Index))
+                    {
+                        cachedChoices.Add(choice.Index, new());
+                        cachedChoices[choice.Index].Add(choice);
+
+                        var result = new TextStreamingResult(completions, cachedChoices[choice.Index]);
+                        results.Add(result);
+                    }
+                    else
+                    {
+                        cachedChoices[choice.Index].Add(choice);
+                    }
+
+                    if (!streamingStarted)
+                    {
+                        streamingStarted = true;
+                    }
+                }
+            }
+
+            // Updates all results to flag end of stream
+            streamingEnded = true;
+            foreach (var result in results)
+            {
+                result.EndOfStream();
+            }
```

Comment:
What happens if this throws an exception? How is that failure communicated back to the user?  Does the system hang because the results were never fully populated?